src/files-and-videos/videos-page/UploadTrackerModal.jsx

```diff
@@ -0,0 +1,132 @@
+}) => {
+  const intl = useIntl();
+  const videosPagePath = '';
+  const { uploadData, uploadCount } = currentUploadingIdsRef;
+
+  return (
+    <ModalDialog
+      title={intl.formatMessage(messages.videoUploadTrackerModalTitle)}
+      isOpen={isUploadTrackerOpen}
+      onClose={handleUploadCancel}
+      isBlocking
+      hasCloseButton={false}
+      size="lg"
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title className="mb-3">
+          {intl.formatMessage(messages.videoUploadTrackerModalTitle)}
+        </ModalDialog.Title>
+        <Alert
+          variant="warning"
+          icon={WarningFilled}
+        >
+          <Alert.Heading>
+            {intl.formatMessage(messages.videoUploadTrackerAlertTitle)}
+          </Alert.Heading>
+          {intl.formatMessage(messages.videoUploadTrackerAlertBodyMessage)}
+          <div className="mt-3">
+            <span className="font-weight-bold">
+              {intl.formatMessage(messages.videoUploadTrackerAlertEditMessage)}
+            </span>
+            <Hyperlink
+              className="ml-2"
+              destination={videosPagePath}
+              target="_blank"
+              content={intl.formatMessage(messages.videoUploadTrackerAlertEditHyperlinkLabel)}
+            />
+          </div>
+        </Alert>
+        <div className="my-4 text-primary-500">
+          {intl.formatMessage(
+            messages.videoUploadTrackerModalBody,
+            { uploadCount },
+          )}
+        </div>
+      </ModalDialog.Header>
+      <Scrollable>
+        <ModalDialog.Body>
+          <div role="list" className="text-primary-500">
+            {Object.entries(uploadData).map(([id, video], index) => {
+              const bulletNumber = `${index + 1}. `;
+              const getIcon = () => {
+                switch (video.status) {
+                case RequestStatus.SUCCESSFUL:
+                  return (<Icon src={Check} />);
+                case RequestStatus.FAILED:
+                  return (<Icon src={ErrorOutline} />);
+                default:
+                  return (<div style={{ width: '24px' }} />);
+                }
+              };
+              return (
+                <Stack role="listitem" gap={2} direction="horizontal" className="mb-3 small" key={id}>
+                  <span>{bulletNumber}</span>
+                  <div className="col-5 pl-0">
+                    <Truncate>
+                      {video?.name}
+                    </Truncate>
+                  </div>
+                  <div className="col-6 p-0">
+                    {video.status === RequestStatus.FAILED ? (
+                      <span className="row m-0 justify-content-end font-weight-bold">
+                        {video.status.toUpperCase()}
+                      </span>
+                    ) : (
+                      <ProgressBar now={video.uploadPercentage} variant="info" />
+                    )}
+                  </div>
+                  {getIcon()}
```

Comment:
This looks to me like an instance of the known `renderThing()` antipattern.  Calling a `getIcon()` function just looks like an alternative way to a React component. How about instead of having a `getIcon()` function having a `RequestStatusIcon` component that does exactly the same? That's much more the React way.